### PR TITLE
feat(crypto-js): Migrate tests and polish the API

### PR DESCRIPTION
**Pull request by Hywan**
The idea of this PR is to migrate tests from `node:test` to `jest`, and to polish the API to get something similar to `matrix-sdk-crypto-nodejs`. Another goal is to remove the `Makefile` and to replace it by `npm run <script>` because it's easier for the potential users.

Finally, the migrated tests are now run in the CI.

### CODE REVIEW

File: crates/matrix-sdk-crypto-js/Cargo.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ docsrs = []
 
 [dependencies]
 matrix-sdk-crypto = { version = "0.5.0", path = "../matrix-sdk-crypto" }
+matrix-sdk-common = { version = "0.5.0", path = "../matrix-sdk-common" }
```
[the remark this answers is not in the document]
Comment (Hywan) [reply crates/matrix-sdk-crypto-js/Cargo.toml:31]: :+1:

File: crates/matrix-sdk-crypto-js/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# `matrix-sdk-crypto-js`
+
+Welcome to the [WebAssembly] + JavaScript binding for the Rust
+[`matrix-sdk-crypto`] library! WebAssembly can run anywhere, but this
+binding is designed to run on a JavaScript host. This binding is part
+of the [`matrix-rust-sdk`] project, which is a library implementation
+of a [Matrix] client-server.
+
+`matrix-sdk-crypto-js` is a no-network-IO implementation of a state
```
[the remark this answers is not in the document]
Comment (Hywan) [reply crates/matrix-sdk-crypto-js/README.md:9]: :+1:

File: crates/matrix-sdk-crypto-js/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+machine, named `OlmMachine`, that handles E2EE ([End-to-End
+Encryption](https://en.wikipedia.org/wiki/End-to-end_encryption)) for
+[Matrix] clients.
+
+## Usage
+
+This WebAssembly binding is written in [Rust]. To build this binding, you
```
[the remark this answers is not in the document]
Comment (Hywan) [reply crates/matrix-sdk-crypto-js/README.md:16]: :+1:

**Review by jplatte — APPROVED** [review]
Nitpick about the commit message: "feedback" has no plural. Changes look good 👍🏼

**Pull request merged**